### PR TITLE
[Cache] Introduce a new CacheInterface

### PR DESCRIPTION
**Pull request by mickaelandrieu**
Added an interface instead of use `AbstractCache` as the contract for each Cache adapter.

ping @crouillon

### CODE REVIEW

**Comment by crouillon** [comment]
Good idea, but you'll have to add ExtendedCacheInterface also
The cachelistener should be also changed

File: Cache/CacheInterface.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    public function setContext($context);
+
+    /**
+     * Logs a message on provided level if a logger is defined.
+     *
+     * @param string $level   The log level
+     * @param string $message The message to log
+     * @param array  $context The logging context
+     */
+    public function log($level, $message, array $context);
```
Comment (crouillon) [inline Cache/CacheInterface.php]: no need this public in interface i think

File: Cache/CacheInterface.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    /**
+     * Returns the expiration timestamp.
+     *
+     * @param int $lifetime
+     *
+     * @return int
+     */
+    public function getExpireTime($lifetime);
```
Comment (crouillon) [inline Cache/CacheInterface.php]: no need this public in interface i think

File: Cache/CacheInterface.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    /**
+     * Control the lifetime against min and max lifetime if provided.
+     *
+     * @param int $lifetime
+     *
+     * @return int
+     */
+    public function getControledLifetime($lifetime);
```
Comment (crouillon) [inline Cache/CacheInterface.php]: no need this public in interface i think

File: Cache/CacheInterface.php (hunk 1 above)
Comment (eric-chau) [inline Cache/CacheInterface.php]: Agree, the action to log something is up to the developer.

**Comment by mickaelandrieu** [comment]
thanks for review @crouillon

File: Cache/CacheExtendedInterface.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * Interface for BackBee Cache Tag & Expire features
+ *
+ * @category    BackBee
+ *
+ * @copyright   Lp digital system
+ * @author      Mickaël Andrieu <mickael.andrieu@lp-digital.fr>
+ */
+interface CacheExtendedInterface
```
Comment (crouillon) [inline Cache/CacheExtendedInterface.php]: should extends CacheInterface

File: Event/Listener/CacheListener.php [hunk 5]
```diff
@@ -133,17 +134,21 @@ public function __construct(ApplicationInterface $application, CacheValidator $v
...
         }
 
-        if (true === $this->application->getContainer()->has('cache.page')) {
-            $cache_page = $this->application->getContainer()->get('cache.page');
-            if (true === ($cache_page instanceof AbstractExtendedCache)) {
-                $this->cache_page = $cache_page;
+        if ($this->application->getContainer()->has('cache.page')) {
+            $cachePage = $this->application->getContainer()->get('cache.page');
+            if ($cachePage instanceof CacheInterface
+                && $cachePage instanceof CacheExtendedInterface
```
Comment (crouillon) [inline Event/Listener/CacheListener.php]: same as previously

**Comment by mickaelandrieu** [comment]
yes, will be solved in the next pull request :)

**Pull request merged**